exchange/exchange.go

```diff
@@ -139,20 +139,30 @@ func (e *exchange) HoldAuction(ctx context.Context, bidRequest *openrtb.BidReque
 	conversions := e.currencyConverter.Rates()
 
 	adapterBids, adapterExtra := e.getAllBids(auctionCtx, cleanRequests, aliases, bidAdjustmentFactors, blabels, conversions)
-	bidCategory, adapterBids, err := applyCategoryMapping(ctx, requestExt, adapterBids, *categoriesFetcher, targData)
-	auc := newAuction(adapterBids, len(bidRequest.Imp))
-	if err != nil {
-		return nil, fmt.Errorf("Error in category mapping : %s", err.Error())
+	anyBidsReturned := false
+	for _, seatBid := range adapterBids {
+		if seatBid != nil && len(seatBid.bids) > 0 {
+			anyBidsReturned = true
+			break
+		}
 	}
+	if anyBidsReturned {
+		bidCategory, adapterBids, err := applyCategoryMapping(ctx, requestExt, adapterBids, *categoriesFetcher, targData)
+		auc := newAuction(adapterBids, len(bidRequest.Imp))
+		if err != nil {
```

Comment:
Shouldn't this `err` be checked before the previous line: `auc := newAuction(adapterBids, len(bidRequest.Imp))`? I understand that this PR isn't the one introducing this change and it was like that before but this seems like a small change that would be good to have because we're calling `newAuction` unnecessarily in case of an err from `applyCategoryMapping`.

Reply:
Totally makes sense. Done.